pandas/core/internals/managers.py

```diff
@@ -1673,8 +1673,10 @@ def create_block_manager_from_arrays(arrays, names, axes):
 
 def construction_error(tot_items, block_shape, axes, e=None):
     """ raise a helpful message about our construction """
-    passed = tuple(map(int, [tot_items] + list(block_shape)))
-    implied = tuple(map(int, [len(ax) for ax in axes]))
+    passed = tuple(map(int, list(block_shape) + [tot_items]))
```

Comment:
just do a reverse if ndim == 2 here

Reply:
Probably not a good idea for panels with 3 dimensions (items, major, minor)
with reversed for ndim == 2: (items, major, minor)
current: (major, minor, items)

The current version is the intended behavior, right?

Reply:
I don't care about Panels, they are being removed shortly.